### ISSUE TITLE
Support dispatching triggers to different triggerer instances.

### ISSUE THREAD

**Issue by UnightSun** [apache/airflow#33818]
### Description

Let it say:
We have two celery workers: A and B, they cannot access to each other
worker A can access file A, but can not access file B, 
and worker B can access file B, not file A.
And we need to check **both** file's status and move on.

With sensors, we can set workers A and B to different queue, and specify task's `queue` parameter.
But when we trying to use deferrable operator and triggers, we cannot specify which triggerer it runs on.

### Use case/motivation

`airflow triggerer` can take a parameter like celery worker's queue, and `Operator.defer` or `Operator.__init__` can specify that parameter, make tigger runs on that triggerer (or one of triggerers).

### Are you willing to submit a PR?

- [ ] Yes I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://github.com/apache/airflow/blob/main/CODE_OF_CONDUCT.md)

**Comment by potiuk** [comment]
Yes. I thin it's a good idea and necessary feature to complet multi-tenancy.

**Comment by avkirilishin** [comment]
@UnightSun @potiuk please take a look at this [PR](https://github.com/apache/airflow/pull/36338)
